src/main/kotlin/com/openlattice/chronicle/services/timeusediary/TimeUseDiaryService.kt

```diff
@@ -0,0 +1,188 @@
+                val currentUUID = UUID.fromString(result.getString(TUD_ID.name))
+                if (submissionsByDate[currentDate].isNullOrEmpty()) {
+                    submissionsByDate[currentDate] = mutableSetOf(currentUUID)
+                } else {
+                    submissionsByDate[currentDate]!!.add(currentUUID)
+                }
+            }
+        } catch (e: SQLException) {
+            logger.error("hds error: $e")
+        }
+        return submissionsByDate
+    }
+
+    override fun downloadTimeUseDiaryData(
+        organizationId: UUID,
+        studyId: UUID,
+        participantId: String,
+        type: TimeUseDiaryDownloadDataType,
+        submissionsIds: Set<UUID>
+    ) {
+        TODO("Not yet implemented")
+    }
+
+    /* -------- SQL helpers -------- */
+
+    private fun executeInsertTimeUseDiarySql(
+        connection: Connection,
+        tudId: UUID,
+        organizationId: UUID,
+        studyId: UUID,
+        participantId: String,
+        responses: List<TimeUseDiaryResponse>
+    ) {
+        val preparedStatement = connection.prepareStatement(insertTimeUseDiarySql)
+        var index = 1
+        preparedStatement.setObject(index++, tudId)
+        preparedStatement.setObject(index++, organizationId)
+        preparedStatement.setObject(index++, studyId)
+        preparedStatement.setString(index++, participantId)
+        preparedStatement.setObject(index, objectMapper.writeValueAsString(responses), Types.OTHER)
+        preparedStatement.execute()
+    }
+
+    private fun executeGetSubmissionByDateSql(
+        connection: Connection,
+        organizationId: UUID,
+        studyId: UUID,
+        participantId: String,
+        startDate: OffsetDateTime,
+        endDate: OffsetDateTime
+    ): ResultSet {
+        val preparedStatement = connection.prepareStatement(getSubmissionsByDateSql)
+        var index = 1
+        preparedStatement.setObject(index++, organizationId)
+        preparedStatement.setObject(index++, studyId)
+        preparedStatement.setString(index++, participantId)
+        preparedStatement.setObject(index++, startDate)
+        preparedStatement.setObject(index, endDate)
+        return preparedStatement.executeQuery()
+    }
+
+    /**
+     * SQL String to create a [java.sql.PreparedStatement] to submit a study response for a participant
+     *
+     * @param tudId             Identifies the Time Use Diary submission
+     * @param organizationId    Identifies the organization from which to retrieve submissions
+     * @param studyId           Identifies the study from which to retrieve submissions
+     * @param participantId     Identifies the participant for whom to retrieve submissions
+     * @param responses         List of survey responses for a Time Use Diary study
+     */
+    private val insertTimeUseDiarySql = """
+            INSERT INTO ${TIME_USE_DIARY_SUBMISSION.name} 
+            VALUES ( ?, ?, ?, ?, '${LocalDateTime.now()}', ? )
```

Comment:
A quick comment here. LocalDateTime.now() called on our servers will always be UTC time. If the timezone of a submission is important then you have to collect timezone on the front end and pass it all the way through (or at least lat/lon to be able to triangulate timezone of submission). In this case you can use OffsetDateTime.now() or even more natively just the now function i.e ```VALUES (? , ? ,? ,? , now(), ?)```

Reply:
Given that timezone is not important for submissions, I'll update this to use the native now() function as suggested.